src/sentry/integrations/jira/integration.py

```diff
@@ -398,23 +398,32 @@ def get_issue_type_meta(self, issue_type, meta):
 
         return issue_type_meta
 
-    def get_create_issue_config(self, group, **kwargs):
-        kwargs['link_referrer'] = 'jira_integration'
-        fields = super(JiraIntegration, self).get_create_issue_config(group, **kwargs)
-        params = kwargs.get('params', {})
-
-        defaults = self.get_project_defaults(group.project_id)
-        project_id = params.get('project', defaults.get('project'))
-
-        client = self.get_client()
-
-        # If we don't have a jira project selected, fetch the first project
-        # This avoids a potentially very expensive API call to fetch issue
-        # create configuration for *all* projects.
-        jira_projects = client.get_projects_list()
-        if not project_id and len(jira_projects):
-            project_id = jira_projects[0]['id']
+    def get_issue_create_meta(self, client, project_id, jira_projects):
+        if project_id:
+            return self.fetch_issue_create_meta(client, project_id)
+
+        # If we don't have a jira projectid, iterate all projects and find
+        # the first project that has metadata. We only want one project as getting
+        # all project metadata is expensive and wasteful. In the first run experience,
+        # the user won't have a 'last used' project id so we need to iterate available
+        # projects until we find one that we can get metadata for.
+        if len(jira_projects):
+            attempts = 0
+            for fallback in jira_projects:
```

Comment:
Does it make sense to fetch these in batches? Like not the entire org's worth, but it might be a happy medium to fetch a small number at a time?

Reply:
Batches could work well if there are more than 2 attempts. I'm going to watch the log messages created over the weekend and will make a change if I see more than 3 requests happening for multiple customers.